glommio/src/executor.rs

```diff
@@ -559,6 +559,276 @@ impl Default for LocalExecutorBuilder {
+    /// details.  The setting is applied to all executors in the pool.
+    pub fn io_memory(mut self, io_memory: usize) -> Self {
+        self.io_memory = io_memory;
+        self
+    }
+
+    /// Please see documentation under [`LocalExecutorBuilder::preempt_timer`]
+    /// for details.  The setting is applied to all executors in the pool.
+    pub fn preempt_timer(mut self, dur: Duration) -> Self {
+        self.preempt_timer_duration = dur;
+        self
+    }
+
+    /// This method sets the [`Placement`] policy by which [`LocalExecutor`]s
+    /// are bound to the machine's hardware topology (i.e. which CPUs to
+    /// use).  The default is [`Placement::Unbound`].
+    pub fn placement(mut self, p: Placement) -> Self {
+        self.placement = p;
+        self
+    }
+
+    /// A method that generates a [`CPUSet`] according to the provided
+    /// [`Placement`] policy; sequential calls may generate different sets
+    /// depending on the [`Placement`]
+    fn make_cpu_set(&self) -> CPUSet {
+        match self.placement {
+            Placement::Unbound => CPUSet(None),
+        }
+    }
+
+    /// Spawn a pool of [`LocalExecutor`]s in a new thread according to the
+    /// [`Placement`] policy, which is `Unbound` by default.
+    ///
+    /// This method is the pool equivalent of [`LocalExecutorBuilder::spawn`].
+    ///
+    /// It takes a closure `fut_gen` which will be executed on each new thread
+    /// to obtain the [`Future`] to be executed there.  The `fut_gen`
+    /// closure is passed two arguments: a `usize` and a [`CPUSet`].  The
+    /// `usize` provides a sequential id of the threads created by this
+    /// [`LocalExecutorPoolBuilder`] (i.e. numbered in the range `0..nr_shards`
+    /// where `nr_shards` is the number of threads created).  The `CPUSet`
+    /// provides a set of CPUs on which the thread may execute.  In the
+    /// `[Placement::Unbound]` case, this `CPUSet` is not restricted and
+    /// simply holds an `Option::None`.
+    ///
+    /// If the [`Future`] to be executed on each thread needs to customized, the
+    /// `fut_gen` closure may use its two arguments
+    ///
+    /// # Panics
+    ///
+    /// The newly spawned thread panics if creating the executor fails. If you
+    /// need more fine-grained error handling consider initializing those
+    /// entities manually.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use glommio::LocalExecutorPoolBuilder;
+    ///
+    /// let handles = LocalExecutorPoolBuilder::new(4).on_all_shards(|_id, _cpu_set| async move {
+    ///     println!("hello");
+    /// });
+    ///
+    /// handles.join_all();
+    /// ```
+    #[must_use = "This spawns executors on multiple threads, so you may need to call \
+                  `PoolThreadHandles::join_all()` to keep the main thread alive"]
+    pub fn on_all_shards<G, F, T>(self, fut_gen: G) -> PoolThreadHandles
+    where
+        G: FnOnce(usize, CPUSet) -> F + Clone + Send + 'static,
```

Comment:
what is your intention by passing the `cpuset` inside the function for `on_all_shards` ?

Reply:
This is purely for users' convenience / information.  It would allow the user to modify execution based on where the thread will be running (e.g. maybe a particular package/socket has a larger cache so the user might take advantage of that in some way, or threads running on the same core have some preferred work-stealing arrangement, or the user has some method of detecting panics and would use this information to quickly restart a thread that went down ... just spitballing).  I can remove this if you believe it just clutters the interface.

Reply:
Without an example of how this would be used I think it is clutter.